core-bundle/src/Framework/ContaoFramework.php

```diff
@@ -148,17 +146,11 @@ public function createInstance($class, $args = [])
      *
      * @param class-string<T> $class
      *
-     * @return T
-     *
-     * @phpstan-return Adapter<T>
+     * @return Adapter<T>
```

Comment:
I had it this way on purpose, because without the `@return T`, PHPStorm does not understand it:

<img width="701" alt="" src="https://user-images.githubusercontent.com/1192057/143821801-b98c6329-5678-4c02-9015-8a9941b0e844.png">

So I'd prefer if we could revert this, even though it is actually correct.

Reply:
I changed it to

```
@return Adapter<T>&T
@phpstan-return Adapter<T>
```

now in the last commit. This seems to be working for PHPStorm.